### ISSUE TITLE
weird issue : top layer gets ruined (too deep) but bottom layer is fine

### ISSUE THREAD

**Issue by phil123456** [vlachoudis/bCNC#771]
Hi,

first I grave the bottom layer, then everytime I do the top layer, for some weird reason it is graved too deep but I probe the whole area the same way as for the bottom layer and before any engraving

how could I solve this ?

thanks

**Comment by masterleo** [comment]
Hello, could you take pictures and include screen shots please ? What is your OS ? What is the version of BCNC ? We need informations  to solve bugs ;) What is the machine you use ?

**Comment by phil123456** [comment]
Windows 7
I cannot find the version I use...just tried the last one, there is an unresolved bug in it (Key error something)
I use a chinese CNC3018 .... I cant take screeenshots right now, but twhile one side is routed correctly, the other one is barely touched by the spindle bit

I tried to reproduce it with a simbple example, and it routed fine.I'll try with a more complex board, if I can reproduce I'll send you all the files I created

regards

**Comment by MARIOBASZ** [comment]
It's a weird mistake. Can Autolevel have influence?

**Comment by phil123456** [comment]
here are the files and pictures

the second (bottom) tap file must have been updated with the autolevel scan I think, not the first one

![20181026_203134](https://user-images.githubusercontent.com/13568652/47585810-b4e72300-d95e-11e8-8936-13cd5e8101ab.jpg)
![20181026_203145](https://user-images.githubusercontent.com/13568652/47585815-ba446d80-d95e-11e8-9f51-faf4b683a8d6.jpg)

[bCNC_issue.zip](https://github.com/vlachoudis/bCNC/files/2520434/bCNC_issue.zip)

one clearcly see the issue on the pictures

**Comment by Harvie** [comment]
You have to do autolevel for each side separately, every time you mount your PCB to wasteboard.
Also note that if you save the autolevel data under same name as gcode, it will get autoloaded.
Isn't this your problem?

**Comment by phil123456** [comment]
I do autolevel for each side, obviously :-)

**Comment by Harvie** [comment]
So i don't get it. From what you say bCNC is have no way to know which side is milled.
Maybe your board is flexing and once cut it flexes even much? The woodscrews you use might bend and pre-tension the board. Try to use double sided tape to support it across the whole board. Make sure that there is no debris from previous cutting under the board. Also try to lower the probing speed 10 times to see if makes a difference. Maybe the copper on one side of the board is thicker than on another... Maybe there's some backlash or flex in Z, which gets pretensioned while you do first cut. What kind of spindle do you use? Does it have any play in Z (or any other) axis? (loose bearings, so you can move milling but up and down a bit?) PCB engraving is EXTREMELY sensitive for any deviation in Z and if you have play in your machine, the autoleveling just might not be enough. Maybe the bit vibrates up and down while rotating and during autolevel procedure it is not.

**Comment by phil123456** [comment]
> So i don't get it. From what you say bCNC have no way to know which side is milled.

Could'n agree more, makes no sense so far

> Try to use double sided tape to support it across the whole board instead of screws 

Yep, thought about that, I'll give it a try

> Make sure that there is no debris from previous cutting under the board

Checked

> Also try to lower the probing speed 10 times to see if makes a difference

hmm, did'nt know about that, I'll see whe I can find that setting

> What kind of spindle do you use

it's the motor that came with the cnc, there is no z movement on the axle at rest, alredy tried
I also use an er11 for holding the bit, and a fresh carving tip everytime

I'll check the whole montage again, good idea

thanks for all these tips dude ;-)

**Comment by Harvie** [comment]
>> Also try to lower the probing speed 10 times to see if makes a difference

> I'll see where I can find that setting

![image](https://user-images.githubusercontent.com/276504/47785674-b7a09a00-dd09-11e8-8137-cb104ae34e06.png)

**Comment by phil123456** [comment]
I just remembered that everytime I do the scan, I click "reset" otherwise the spindle does not start, I also have to start the spindle manualy

but when I click start, as the spindle is running, the thing goes back to center while beeing really close to the board, instead of setting the motor at high position first...it creates sometimes a really thin line on the pcb going from the start scan point to the center point

not a big deal at first, but that alone maybe damages the bit before even starting

**Comment by Harvie** [comment]
> it creates sometimes a really thin line on the pcb going from the start scan point to the center point

Do you have latest git version of bCNC? I think i've fixed this some time ago. But it all boils down to how you generate your g-code. If you use "cut" operation in bCNC it should not happen. Also be sure to disable other blocks and jog Z up before you hit start.

**Comment by phil123456** [comment]
I never use python 3 :-)

**Comment by Harvie** [comment]
> I never use python 3 :-)

![image](https://user-images.githubusercontent.com/276504/48055469-f625d100-e1af-11e8-934a-15cf98387adf.png)

what is this then?

**Comment by phil123456** [comment]
I dont get it, I am sure I use python 2.7, I'll check when @ home

**Comment by Harvie** [comment]
Note that you can have multiple python versions installed. That's not really uncommon.
Also in such case you need to make sure, that you install numpy module to the right python installation using pip2. There were people reporting problems with numpy module missing, but they've installed numpy to python3 instead.

Also note that there's python3 bCNC port on the way. Some stuff is still broken, but there will be transition to python3 in few months.

**Comment by phil123456** [comment]
right, so I just tried to cnc a pcb and it's even worst than before
not only the bug is not resolved, after a scan the bit is so close to the pcb it traces a big line
but now even if I lower 0.1 unit the z axis befor starting, it still engraves half of the traces

I am looking for another more decent software

**Comment by Harvie** [comment]
> bCNC definetely treats bottom and top layer differently

AFAIK bCNC has no concept of layers. So there's no way it will tell the difference. Last piece of your toolchain that can tell layers is CAM or g-code generator, that you use.

> I am now looking for another more decent software

Please let me know what are the results with other software. I can reccomend you to try UGS or newer UGS platform. If you will get better (or exactly same) results with UGS or similar software it will be great starting point to fixing this bug (whether it lays in bCNC or elsewhere).

**Comment by Harvie** [comment]
> after a scan the bit is so close to the pcb it traces a big line

Do you know that you probably should not start right after the scan? You probably should retract and/or rezero. Does your g-code contain Z coordinates? At least something like Z0 at beginning.
